datadog_sync/utils/resources_handler.py

```diff
@@ -281,21 +286,70 @@ async def _pre_apply_hook_cb(self, resource_type: str) -> None:
     async def run_sorter(self):
         loop = asyncio.get_event_loop()
         while await loop.run_in_executor(None, self.sorter.is_active):
-            for _id in self.sorter.get_ready():
-                if _id not in self.resources_manager.all_resources_to_type:
+            for node in self.sorter.get_ready():
+                if node[1] not in self.config.state.source[node[0]]:
                     # at this point, we already attempted to import missing resources
                     # so mark the node as complete and continue
-                    self.sorter.done(_id)
+                    self.sorter.done(node)
                     continue
-                await self.worker.work_queue.put((self.resources_manager.all_resources_to_type[_id], _id))
+                await self.worker.work_queue.put(node)
             await asyncio.sleep(0)
 
+    def get_dependency_graph(self) -> Tuple[Dict[Tuple[str, str], List[Tuple[str, str]]], Set[Tuple[str, str]]]:
+        """Build the dependency graph for all resources.
+
+        Returns:
+            Tuple[Dict[Tuple[str, str], List[Tuple[str, str]]], Set[Tuple[str, str]]]: Returns
+            a tuple of the dependency graph and missing resources.
+        """
+        dependency_graph = {}
+        missing_resources = set()
+
+        for resource_type, _id in self.config.state.get_all_resources(self.config.resources_arg).keys():
+            deps, missing = self._resource_connections(resource_type, _id)
+            dependency_graph[(resource_type, _id)] = deps
+            missing_resources.update(missing)
+
+        return dependency_graph, missing_resources
+
+    def _resource_connections(self, resource_type: str, _id: str) -> Tuple[Set[Tuple[str, str]], Set[Tuple[str, str]]]:
```

Comment:
We should probably document what this is returning. It says it's returning `failed_connections, missing_resources` but `failed_connections` is also treated as `deps` above.